subprojects/docs/src/docs/userguide/dependencyManagementTerminology.adoc

```diff
@@ -0,0 +1,44 @@
+
+[[sub:terminology_module]]
+Module::
+
+A piece of software that evolves over time e.g. link:https://github.com/google/guava[Google Guava]. Every module has a name. Each release of a module is optimally represented by a <<sub:terminology_version,version>>. For convenient consumption, modules can be hosted in a <<sub:terminology_repository,repository>>.
+
+[[sub:terminology_version]]
+Version::
+
+A <<sub:terminology_module,module>> indicates its revision with the help of a version. Each module version contains a distinct set of changes. In practice there's no limitation to the scheme of the version. Timestamps, numbers, special suffixes like GA are all allowed identifiers. The most widely-used versioning strategy is link:https://semver.org/[semantic versioning]. Semantic versioning is represented by the scheme _major_._minor_._patch_ e.g. `1.4.15`. Each part conveys a special meaning about the backwards-compatibility of the release.
+
+[[sub:terminology_dependency]]
+Dependency::
+
+A dependency is defined as a versioned or unversioned module. Dependencies may provide <<sub:terminology_dependency_metadata,metadata>> to indicate that they depend on other dependencies required for it work properly, so-called <<sub:terminology_transitive_dependency,transitive dependencies>>. For more information, see <<declaring_dependencies>>. Gradle's dependency resolution process can apply constraints declared in the build script to select a specific version.
+
+[[sub:terminology_transitive_dependency]]
+Transitive dependency::
+
+A transitive dependency is a dependency required by another dependency to work properly. Dependencies hosted on a <<sec:terminology_repository,repository>> can provide <<sub:terminology_dependency_metadata,metadata>> declaring transitive dependencies. By default, Gradle resolves transitive dependencies if defined by metadata. The behavior is highly customizable. For more information, see <<managing_transitive_dependencies>>.
+
+[[sub:terminology_dependency_metadata]]
+Dependency metadata::
+
+Dependencies can provide metadata. Metadata is the data that describes the dependency in more detail e.g. the coordinates for locating it in a repository, information about the project or required <<sub:terminology_transitive_dependency,transitive dependencies>>. In Maven the metadata file is called `pom.xml`, in Ivy it is called `ivy.xml`.
+
+[[sub:terminology_configuration]]
+Configuration::
+
+A configuration represents the scope in a Gradle project a set of <<sub:terminology_dependency,dependencies>> should apply to e.g. the Guava dependency should be used to compile source code. The word "configuration" is an overloaded term and has a different meaning outside of the context of dependency management. For more information, see <<sub:scope_of_dependency_configurations>>.
```

Comment:
:x: I find this definition unclear. I would rather say something like:

> A configuration is a named set of dependencies grouped together for a specific goal: For example the `implementation` configuration represents the set of dependencies required to compile a project.

Reply:
:+1: I agree and I like Cedrics proposal. I would replace the first sentence with it.

I  also don't like the _The word "configuration" is an overloaded..._ very much. Sounds like this is part of the definition which is confusing. And other terms are probably also overloaded (module? dependency?). I would be fine with removing it.
If you want to keep it, maybe move it to the end and add "Note:" in front of it. To make it clear that this is a hint about the usage of the term in a different context.

Reply:
Both @melix and @jjohannes suggestions are 💯  super-legit IMO.

Reply:
done. I will probably keep the note. I am sure more than 10 people I talked to outside of Gradle made the statement that "configuration" is confusing. I am going to keep it in for now.